ui/src/kapacitor/containers/KapacitorPage.js

```diff
@@ -71,6 +71,23 @@ class KapacitorPage extends Component {
     const {addFlashMessage, source, params, router} = this.props
     const {kapacitor} = this.state
 
+    let doesKapacitorNameAlreadyExist = false
+
+    if (source && source.kapacitors) {
+      source.kapacitors.map(existingKapacitor => {
+        if (existingKapacitor.name === kapacitor.name) {
+          doesKapacitorNameAlreadyExist = true
+        }
+      })
+    }
+    if (doesKapacitorNameAlreadyExist === true) {
+      addFlashMessage({
+        type: 'error',
+        text: `There is already a Kapacitor configuration named "${kapacitor.name}"`,
+      })
+      return
+    }
+
```

Comment:
Also, I looked to see what the source empty state looks like. It appears as though we can assume that a `source` will exist on this page, along with an empty `kapacitors` array if there are no kapacitors. So you can probably remove that check.

Though we should now make explicit in the `propTypes` that we require a kapacitor array. Make the source required, and add an array of kapacitors:
```js
  source: shape({
    id: string.isRequired,
    url: string.isRequired,
    kapacitors: array.isRequired,
  }).isRequired,
```